### ISSUE TITLE
G final en autocomplétion

### ISSUE THREAD

**Issue by cquest** [addok/addok#465]
Exemple: VICTOR HUG -> VIKTOR U

Le G final est supprimé par https://github.com/addok/addok-fr/blob/master/addok_fr/utils.py#L34

Dans le cas d'un mot complet c'est compréhensible, mais pour l'autocomplétion cela pose un problème...

**Comment by yohanboniface** [comment]
Oui, c'est le cas pour toutes les lettres qui sont traitées en fonction de ce qui suit.
addok-trigrams est une possible réponse pour se débarrasser de ce genre de règle.

**Comment by cquest** [comment]
Le problème c'est que la tokenisation est faite avant le fuzzy et l'autocomplete, et empêche donc l'autocomplete de proposer des tokens pertinents (UG -> UGO)

Ne devrait-on pas changer l'ordre du traitement ou revoir la façon de générer les tokens d'autocomplete ?
Ce n'est sûrement pas le seul cas de ce type.

**Comment by cquest** [comment]
Autre cas identifié: 32 RUE DE MOG (pour MOGADOR)

**Comment by Flaburgan** [comment]
J'ai pas mal de problèmes similaires avec par exemple `pontch` ou `pontc` qui devient `pontk` alors que je cherche à trouver `pontcharra` en autocomplete. Est-ce que  addok-fr ne devrait pas être désactivé lorsque l'API est appelé avec `autocomplete` à 1 ?

**Comment by yohanboniface** [comment]
cf #480 aussi

Une des pistes qui sortent de #492 c'est effectivement d'avoir  un mode `autocomplete` dans addok-fr qui ne touche pas aux dernières lettres, à appeler uniquement pour le dernier token quand on calcule les tokens d'autocomplétion.
